lib/src/configuration.dart

```diff
@@ -104,6 +110,14 @@ class Configuration {
   /// if it was created with an older version of the [Realm] library.
   /// An exception will be thrown if a file format upgrade is required.
   final bool disableFormatUpgrade;
+  
+  // A function that will be executed each time a Realm is opened.
```

Comment:
We should be checking the .NET SDK doc comments. This one diverges from what this really is doing. https://www.mongodb.com/docs/realm-sdks/dotnet/latest/reference/Realms.RealmConfiguration.ShouldCompactDelegate.html
It is not executed on each realm open but only the first time during the life time of the process.

Reply:
Updated from JS SDK.